### PR TITLE
Don't ignore duplicate parts written to replicas

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
When duplicate block is written to replica where it does not exist locally (has not been fetched from replicas), don't ignore it and write locally to achieve the same effect as if it was successfully replicated.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
TODO:
- test when part already exists locally;
- test when duplicate was suddenly appeared at the moment of final transaction;
- update quorum.

**Comment by alexey-milovidov** [comment]
> Yandex third-party checks (only for Yandex employees) 

Something broken with sync (it reports about already fixed issue).

**Comment by alexey-milovidov** [comment]
All good.

**Comment by alexey-milovidov** [comment]
Integration test fail sometimes, need to investigate.

File: src/Storages/MergeTree/MergeTreeData.h [hunk 1]
```diff
@@ -440,6 +442,9 @@ class MergeTreeData : public IStorage
             MutableDataPartPtr & part, SimpleIncrement * increment, Transaction * out_transaction, DataPartsLock & lock,
             DataPartsVector * out_covered_parts = nullptr);
 
+
+    void removePartsFromWorkingSetImmediatelyAndSetTemporaryState(const DataPartsVector & remove);
```
Comment (alesapin) [inline src/Storages/MergeTree/MergeTreeData.h:446]: Need to add comments.

**Comment by alesapin** [comment]
Test already finished https://clickhouse-test-reports.s3.yandex.net/11684/c8a5d9e0a2da0a30b2f5881a727126a2a5ac9530/functional_stateless_tests_(thread).html.

**Comment by alesapin** [comment]
performance not affected.

**Pull request merged**